app/scripts/modules/core/src/function/Functions.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import { Debounce } from 'lodash-decorators';
+import { Subscription } from 'rxjs';
+import { Application } from 'core/application/application.model';
+import { FilterTags, IFilterTag } from 'core/filterModel/FilterTags';
+import { IFunctionGroup } from 'core/domain';
+import { FunctionState } from 'core/state';
+import { FunctionPod } from './FunctionPod';
+import { Spinner } from 'core/widgets/spinners/Spinner';
+
+export interface IFunctionsProps {
+  app: Application;
+}
+
+export interface IFunctionsState {
+  initialized: boolean;
+  groups: IFunctionGroup[];
+  tags: IFilterTag[];
+}
+
+export class Functions extends React.Component<IFunctionsProps, IFunctionsState> {
+  private groupsUpdatedListener: Subscription;
+  private functionsRefreshUnsubscribe: () => any;
```

Comment:
It seems like this returns void rather than any, but maybe I am interpreting the function incorrectly

Reply:
the definition of that function maps to 
```
this.functionsRefreshUnsubscribe = app
      .getDataSource('functions')
      .onRefresh(null, () => this.updateFunctionGroups());
```
the `onRefresh` returns `any`